omod/src/main/java/org/openmrs/module/openconceptlab/web/rest/resources/ImportResource.java

```diff
@@ -173,4 +178,27 @@ private UpdateScheduler getUpdateScheduler() {
     private static ImportService getImportService() {
         return Context.getService(ImportService.class);
     }
+
+    @Override
+    public Object upload(MultipartFile multipartFile, RequestContext requestContext) throws ResponseException, IOException {
+
+        ImportService importService = getImportService();
+
+        Importer importer = Context.getRegisteredComponent("openconceptlab.importer", Importer.class);
+        importer.setImportService(importService);
+
+        File file = new File(multipartFile.getOriginalFilename());
+        multipartFile.transferTo(file);
+        ZipFile zipFile = new ZipFile(file);
+
+        importer.run(zipFile);
```

Comment:
We shouldn't be importing anything synchronously. See UpdateScheduler.scheduleNow(). You need to find a way to pass the file to be imported along to the scheduler for the importer to process it.